Duplicati/Library/Utility/ReadLimitLengthStream.cs

```diff
@@ -4,24 +4,66 @@
 using System.Threading.Tasks;
 
 namespace Duplicati.Library.Utility
-{
-    // StreamReadLimitLengthWrapper() based on code from Matt Smith (Oct 26 '15 at 20:38)
-    // https://stackoverflow.com/questions/33354822/how-to-set-length-in-stream-without-truncated
-
+{
+    // StreamReadLimitLengthWrapper() based on code from Matt Smith (Oct 26 '15 at 20:38)
+    // https://stackoverflow.com/questions/33354822/how-to-set-length-in-stream-without-truncated
+
+    /// <summary>
+    /// This class wraps a Stream but only allows reading a limited number of bytes from the underlying stream.
+    /// This can be used to create split a source stream into multiple smaller buffers that can be used as independent streams.

```

Comment:
> ... that can be used as independent streams.

I guess we should still be careful with how we use this class.  Given a base stream, I think a second instance of `ReadLimitLengthStream` should not be instantiated until existing instances with the same base stream have been disposed.

For example, I believe the following test will fail:
```
MemoryStream baseStream = new MemoryStream(Enumerable.Range(0, 10).Select(i => (byte) i).ToArray());
Stream stream1 = new ReadLimitLengthStream(baseStream, 0, 5);
Stream stream2 = new ReadLimitLengthStream(baseStream, 5, 5);
Assert.AreEqual(0, stream1.Position);
```

Reply:
Yes, though I think that applies almost any time you wrap a stream, since position is an underlying piece of state that all wrappers will be fighting over.

Reply:
To add to that, even if you have multiple ReadLengthLimitStreams active, so long as you aren't using them concurrently in different threads, they should give reasonable behavior (at least, they shouldn't read things outside their own regions) since they look at and clamp the underlying stream's position. Whoever is using it might need seek to the place they want to use it each time (since the position will keep jumping around), but besides that it should pretty much work.

Reply:
Yes, that's true.  The position jumping around sounds a bit risky, and I'm not sure if seeking it will always work (e.g., if one thread sets the position while another is in the middle of a `CopyTo`).  However, like you said this risk just comes with wrapping streams in general.

Reply:
I'll update the description comment a bit as well to warn about this.